lib/crypt.c

```diff
@@ -110,6 +110,15 @@ get_hashfn (const char *setting)
   return 0;
 }
 
+static int
+check_badsalt_chars (const char *setting, size_t set_size)
+{
+  if (strcspn (setting, (const char *) badsalt_chars) != set_size)
+    return 1;
+
+  return 0;
```

Comment:
I think the explicit returns of 1 and 0 don't improve readability for anyone familiar with C. I suggest you return the result of the comparison directly, making this function a one-liner.

Reply:
Addressed in rebased commit.